src/base/bittorrent/session.cpp

```diff
@@ -4064,6 +4069,35 @@ void Session::handleMetadataReceivedAlert(const lt::metadata_received_alert *p)
     }
 }
 
+void Session::handleTorrentPausedAlert(const libtorrent::torrent_paused_alert *p)
+{
+    const InfoHash hash {p->handle.info_hash()};
+
+    if (m_loadedMetadata.contains(hash)) {
+        // Adding preloaded torrent
+        m_loadedMetadata.remove(hash);
+        lt::torrent_handle handle = m_nativeSession->find_torrent(hash);
```

Comment:
since handling alerts are async operations, then it should also check if the handle is still valid.

Reply:
If m_loadedMetadata contains this torrent it is supposed to be existing since we first remove it there ant then delete from the session.

Reply:
Well, maybe not exactly... In any case, the life cycle of such a "hidden" torrent is not too complicated, and there should be no conflicts. If it is there for some reason, it is a clear flaw in the existing logic, and it must be fixed. I'll recheck the corresponding code paths later.

Reply:
Ha, I found an opportunity of race condition in the related code, not here, but in the existing one.
I intend to merge this PR as is and fix this issue in a separate PR.